nemo/collections/asr/parts/mixins/interctc_mixin.py

```diff
@@ -0,0 +1,206 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Dict, List, Tuple
+
+import torch
+
+from nemo.core.classes.mixins import AccessMixin
+
+
+class InterCTCMixin:
+    """Adds utilities for computing interCTC loss from https://arxiv.org/abs/2102.03216.
+
+    To use, make sure encoder accesses `interctc['capture_layers']`
+    property in the AccessMixin and registers interctc/layer_output_X and
+    interctc/layer_length_X for all layers that we want to get loss from.
+    Additionally, specify the following config parameters to set up loss:
+
+        interctc:
+            # can use different values
+            loss_weights: [0.3]
+            apply_at_layers: [8]
+
+    Then call
+
+        * ``setup_interctc`` in the init method
+        * ``self.add_interctc_losses`` after computing regular loss.
+        * ``self.finalize_interctc_metrics(metrics, outputs, prefix="val_")``
+          in the `multi_validation_epoch_end` method.
+        * ``self.finalize_interctc_metrics(metrics, outputs, prefix="test_")``
+          in the `multi_test_epoch_end` method.
+    """
+
```

Comment:
Add getter and setter for a bool variable - is_interctc_enabled() (already added) and set_interctc_enabled(enabled: bool). Instance methods, they check if hasattr() self._interctc_enabled and returns that or sets it explicitly. 

It will allow users to disable it even if present in config which gives more flexibility (say finetuning or want to experiment). Plus it gives better access to whether the class should allow interctc of not even if access is enabled for other use cases (adapter losses, SSL etc)